SPECS/containerized-data-importer/containerized-data-importer.spec

```diff
@@ -32,6 +32,22 @@ BuildRequires:  pkgconfig
 BuildRequires:  rsync
 BuildRequires:  sed
 ExclusiveArch:  x86_64 aarch64
+Requires:       cpio
+Requires:       curl
+Requires:       diffutils
+Requires:       findutils
+Requires:       jq
+Requires:       libnbd
+Requires:       nbdkit
+Requires:       patch
+Requires:       rsync
+Requires:       librsync
+Requires:       qemu-tools
+Requires:       qemu-img
+Requires:       shadow-utils
+Requires:       tar
```

Comment:
Are we sure those are required dependencies? I highly doubt it.
If you attempt running the binary without them, does it not work?

Reply:
This is the [RPM tree](https://github.com/kubevirt/containerized-data-importer/blob/main/rpm/BUILD.bazel) from CDI sources.  Containers, such as cdi-importer and cdi-uploadserver, call sed, tar or qemu-img commands internally in some code paths.  During the initial unit testing of golden containers for platform helm charts, these dependencies got detected during experimentation and included in the docker file.

There are other executables, such as readline, tzdata, gawk, which are seen in the RPM tree hierarchy link above.  We have not hit those errors as yet.  Should we include each and every package in the tree under the Requires directive?

Reply:
@romoh - With the nbdkit packages moved to core, it is time to review this PR.  Please help address above queries.

Reply:
Another question- Is the RPM tree hierarchy reflecting just the tools needed, or does it also include the full closure of dependencies required for the necessary tools? If it's the second option, then we can trim down these lists to just the tools needed and let our RPM requirements handle the dependencies of those tools.

Reply:
> Another question- Is the RPM tree hierarchy reflecting just the tools needed, or does it also include the full closure of dependencies required for the necessary tools? If it's the second option, then we can trim down these lists to just the tools needed and let our RPM requirements handle the dependencies of those tools.

It is the latter.  The RPM tree lists all the libraries, tools and other CLIs.  In the initial check-in for CDI golden containers, I manually sifted through the test failures to figure out reasons for failures.  As there was no data volume import test, we slipped on the nbdkit-server dependency.

Reply:
I think I found where they keep the list of the tools that are actually required, which they use to generate the RPM tree:

https://github.com/kubevirt/containerized-data-importer/blob/main/hack/build/rpm-deps.sh

Mind trimming down these lists to just what's required in this document?

Reply:
This requires two more packages - crypto-policies and codespell - which are absent in Mariner's set of packages.  These two packages also form the base for all the CDI binaries.

Reply:
And there are further sub-dependencies from crypto-policies namely - python3-flake8, java-1.8.0-openjdk-devel and python3-pylint.  So, this is beginning to explode in scope as I gather more and more packages here.